src/python/pants/core/goals/test.py

```diff
@@ -208,6 +280,73 @@ class TestFieldSet(FieldSet, metaclass=ABCMeta):
     __test__ = False
 
 
+_TestFieldSetT = TypeVar("_TestFieldSetT", bound=TestFieldSet)
+
+
+@union
+class TestRequest:
+    """Base class for plugin types wanting to be run as part of `test`.
+
+    Plugins should define a new type which subclasses this type, and set the
+    appropriate class variables.
+    E.g.
+        class DryCleaningRequest(TestRequest):
+            tool_subsystem = DryCleaningSubsystem
+            field_set_type = DryCleaningFieldSet
+
+    Then register the rules which tell Pants about your plugin.
+    E.g.
+        def rules():
+            return [
+                *collect_rules(),
+                *DryCleaningRequest.rules(),
+            ]
+    """
+
+    tool_subsystem: ClassVar[type[SkippableSubsystem]]
+    field_set_type: ClassVar[type[TestFieldSet]]
+    partitioner_type: ClassVar[PartitionerType] = PartitionerType.DEFAULT_ONE_PARTITION_PER_INPUT
+
+    __test__ = False
+
+    @classproperty
+    def tool_name(cls) -> str:
+        return cls.tool_subsystem.options_scope
+
+    @distinct_union_type_per_subclass(in_scope_types=[EnvironmentName])
+    class PartitionRequest(_PartitionFieldSetsRequestBase[_TestFieldSetT]):
+        pass
+
+    @distinct_union_type_per_subclass(in_scope_types=[EnvironmentName])
+    class Batch(_BatchBase[PartitionElementT, PartitionMetadataT]):
+        @property
+        def element(self) -> PartitionElementT:
```

Comment:
```suggestion
        def single_element(self) -> PartitionElementT:
```

To drive the point home.